handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -1576,6 +1582,7 @@ public void handlerAdded(final ChannelHandlerContext ctx) throws Exception {
             } else {
                 applyHandshakeTimeout(null);
             }
+            handshakeStarted = true;
```

Comment:
should we set this before we call handshake, incase some sort of error happens internally?

Reply:
sure why not.

Reply:
same thing in channelActive?